### ISSUE TITLE
[A11y] Search Results - Grid - Search Refinement dropdowns

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1928]
### Operating System

Windows

### Browser

Edge

### Affected page

Search

### Admin user?

Yes

### Issue Description

The "SORT" and "SHOW" dropdowns have issues:

- Neither of these has a correct accessible label - the label is just what's currently on the button. A screen reader user for instance will hear "relevance button collapsed" or worse, "8 button collapsed".
- The widgets seem to be implemented as a menu of links, but they aren't fully coded as such

### Suggested Fix

These drop-down-selection-link widgets are going to be complicated to get right.  They behave nearly like a [menubutton](https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton), but the buttons aren't given a label to describe their purpose; instead the button text describes their current state.  The purpose is elsewhere.  It's tricky to associate the purpose and the current value without just blindly setting the entire label.  But then the experience is weird to say the least, because the button tries to describe two things - its purpose and the current page state.  e.g., if you set the label to "Sort, current value X", it'll be read something like "sort, current value X, menu button, collapsed".  This isn't awful, but it's weird....

Visually these are made to look like a [listbox](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox): they show the current selection and rely on the word next to them to act as a label.  But it's not a list of selectable items for a field on the page; it's a list of links you choose to immediately change the sort/results per page.  To make this a listbox we'd need to redesign this section to be a real form and have an "apply" or "go" button or similar. (And at that point we might just have an easier time with a real `<select>` element). This isn't awful, and may in fact be the right choice, but it would be a pretty significant change.

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

QA depends on the approach taken

**Comment by CGillen** [comment]
Fixed by #2073

**Comment by jechols** [comment]
This is much better, but the menus don't tell a user what the current value is. I don't think that's a hard failure, but I'm not sure. I'm also not sure how we'd fix it without a complete redesign.

I'll give this some thought and leave the ticket alone for now.

**Comment by jechols** [comment]
Opened a new ticket just for the remaining concern: #2158
